### PR TITLE
Add Scroll API for Elastic Client

### PR DESCRIPTION
**Pull request by linonymous**

### CODE REVIEW

File: elastic/client.go [hunk 1]
```diff
@@ -501,3 +501,55 @@ func (p *ClientProvider) UpdateDocumentByQuery(index, query, fields string) ([]b
...
+			p.client.Search.WithBody(&buf),
+			p.client.Search.WithScroll(time.Minute),
+		)
+	} else {
+		res, err = p.client.Scroll(p.client.Scroll.WithScrollID(scrollID), p.client.Scroll.WithScroll(time.Minute))
+	}
+	if err != nil {
+		return err
+	}
+	if res.StatusCode == 200 {
```
Comment (ajinkyan83) [inline elastic/client.go:533]: I think we can use constant for all http status codes.

File: elastic/client.go [hunk 2]
```diff
@@ -501,3 +501,55 @@ func (p *ClientProvider) UpdateDocumentByQuery(index, query, fields string) ([]b
...
+		if err = json.NewDecoder(res.Body).Decode(result); err != nil {
+			return err
+		}
+
+		return nil
+	}
+	if res.IsError() {
+		if res.StatusCode == 404 {
```
Comment (ajinkyan83) [inline elastic/client.go:541]: same, I think we can use constant for all http status codes.

File: elastic/client.go (hunk 1 above)
Comment (linonymous) [reply elastic/client.go]: done.

File: elastic/client.go (hunk 2 above)
Comment (linonymous) [reply elastic/client.go]: done.

**Review by ajinkyan83 — APPROVED** [review]
/lgtm

**Pull request merged**